front/src/common/Map/WarpedMap/core/grids.tsx

```diff
@@ -0,0 +1,245 @@
+        const [[p11]] = outside.geometry.coordinates;
+
+        points[i][direction * j] = p00;
+        points[i][direction * (j + 1)] = p01;
+        points[i + 1][direction * j] = p10;
+        points[i + 1][direction * (j + 1)] = p11;
+      }
+    }
+  }
+
+  return points;
+}
+export function pointsGridToFeature(points: PointsGrid): GridFeature {
+  const grid = featureCollection([]) as GridFeature;
+  const steps = points.length - 1;
+  const stripsPerSide = (Object.keys(points[0]).length - 1) / 2;
+
+  for (let i = 0; i < steps; i++) {
+    for (let direction = -1; direction <= 1; direction += 2) {
+      for (let j = 0; j < stripsPerSide; j++) {
+        const p00 = points[i][direction * j];
+        const p01 = points[i][direction * (j + 1)];
+        const p10 = points[i + 1][direction * j];
+        const p11 = points[i + 1][direction * (j + 1)];
+        grid.features.push(
+          polygon([[p00, p10, p01, p00]], {
+            triangleId: `step:${i}/strip:${direction * (j + 1)}/inside`,
+          }) as Triangle
+        );
+        grid.features.push(
+          polygon([[p11, p10, p01, p11]], {
+            triangleId: `step:${i}/strip:${direction * (j + 1)}/outside`,
+          }) as Triangle
+        );
+      }
+    }
+  }
+
+  return grid;
+}
+
+export function pointsGridToZone(points: PointsGrid): PolygonZone {
+  const firstRow = points[0];
+  const lastRow = points[points.length - 1];
+  const stripsPerSide = (Object.keys(firstRow).length - 1) / 2;
+  const border: Position[] = [];
+
+  // Add first row:
+  for (let i = -stripsPerSide; i <= stripsPerSide; i++) {
+    border.push(points[0][i]);
+  }
+
+  // Add all intermediary rows:
+  for (let i = 1, l = points.length - 1; i < l; i++) {
+    border.push(points[i][stripsPerSide]);
+    border.unshift(points[i][-stripsPerSide]);
+  }
+
+  // Add last row:
+  for (let i = stripsPerSide; i >= -stripsPerSide; i--) {
+    border.push(lastRow[i]);
+  }
+
+  // Close path:
+  border.push(border[0]);
+
+  return {
+    type: 'polygon',
+    points: border,
+  };
+}
+
+/**
+ * This function takes a path, and returns two isomorphic grids:
+ * - The `regular` grid is vertical grid, with each pair of triangles making a
+ *   perfectly regular triangle (considering the earth is flat, ie lat/lng are
+ *   considered as x/y)
+ * - The `warped` grid follows the path
+ * Using these two grids, it becomes possible to project any point from one grid
+ * to the other.
+ */
+export function getGrids(line: Feature<LineString>, params?: { stripsPerSide?: number }): Grids {
+  const l = line.geometry.coordinates.length;
+  if (l <= 2) throw new Error('line must have at least 3 points');
+
+  const stripsPerSide = params?.stripsPerSide || 2;
+
+  const totalLength = length(line);
+  const step = totalLength / (l - 1);
+  const c = center(line);
+  const flatGrid = featureCollection([]) as GridFeature;
+
+  // Generate flat line:
+  const flatLineStart = destination(c, totalLength / 2, 180);
```

Comment:
Why 180 here ? I suppose it is 180 degree, but I don't understand what is happening here
```suggestion
  const flatLineStart = destination(c, totalLength / 2, 180);
```

Reply:
Here, `c` is the center of the new map, and `totalLength` will be its "height", in terms of path lengths.
The starting point of the line that will shape the grid we are looking for is in the bottom of the screen, and the distance between the two points will be `totalLength / 2`.
The bearing must be 180 to go "below", and 0 to go above (but above will be the ending point of the line).

I'll add a comment.